clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1582,7 +1608,11 @@ event_thread_exit(void *drcontext)
         BUF_PTR(data->seg_base) += instru->append_thread_exit(
             BUF_PTR(data->seg_base), dr_get_thread_id(drcontext));
 
-        memtrace(drcontext, true);
+        memtrace(drcontext,
+                 /* If this thread already wrote some data, include its exit even
```

Comment:
Isn't it simpler to just have the `skip_size_cap` arg set to `true`?

Reply:
That is what this code does: it sets `skip_size_cap` to true if any bytes were written.  Am I not understanding the request here?

Reply:
It'd be simpler to set it to `true` **unconditionally**, even if no bytes were written. 

Why do we want to set it to false when `bytes_written==0`? Maybe I'm missing this reason.

Reply:
The comment explains it: we do not want the thread footer if the thread was filtered out (then we'd have a trace with no entries for a particular thread until suddenly there's an exit entry: not what we want).

Reply:
Right, thanks for explaining.